test/operators.jl

```diff
@@ -201,6 +201,37 @@ end
 
 end
 
+@testset "ComposedFunction hash eq" begin
+    @test sin∘cos == sin∘cos
+    @test isequal(sin∘cos, sin∘cos)
+    @test !isequal(sin∘cos, sin∘asin)
+    @test ==(sin∘cos, sin∘cos)
+    @test !==(sin∘cos, sin∘asin)
+    @test hash(sin∘cos) == hash(sin∘cos)
+    @test hash(sin∘cos) != hash(sin∘tan)
+
+    Base.@kwdef mutable struct F
+        isequal::Bool = false
+        eq::Bool = false
+    end
+    function Base.isequal(f1::F, f2::F)
+        f1.isequal
+    end
+    function Base.:(==)(f1::F, f2::F)
+        f1.eq
+    end
+    f2 = F() ∘ F()
+    @test  isequal(F(isequal=true) ∘ F(isequal=true) , f2)
+    @test !isequal(F(isequal=true) ∘ F(isequal=false) , f2)
+    @test !isequal(F(isequal=false) ∘ F(isequal=true) , f2)
+    @test !isequal(F(isequal=false) ∘ F(isequal=false) , f2)
+
+    @test  ==(F(eq=true) ∘ F(eq=true) , f2)
+    @test !==(F(eq=true) ∘ F(eq=false) , f2)
+    @test !==(F(eq=false) ∘ F(eq=true) , f2)
+    @test !==(F(eq=false) ∘ F(eq=false) , f2)
+end
```

Comment:
```suggestion

    @test ((missing ∘ sin) == (missing ∘ sin)) === missing
    @test ((missing ∘ sin) == (missing ∘ cos)) === false
end
```

Not sure, but presumably we'd like to intepret `ComposedFunction` like a collection of functions, thus suppport `missing` as usual.

Reply:
It is a good point, but orthogonal to this PR.

Reply:
To clarify, right now `missing` is not handled specially by composion. 
```julia
julia> missing∘missing
missing ∘ missing
```

Reply:
> orthogonal to this PR.

I don't think so. Right now we have the excuse that `==` for `ComposedFunction` isn't implemented at all, but if we're finally implementing it presumably the handling of `missing` should be thought through.

> right now `missing` is not handled specially by composion.

Composition doesn't need to handle `missing` specially, just `==`. If `ComposedFunction` is interpreted as a sort of collection of functions.

Reply:
You are right, sorry I misread some parenthesis in your test. I read `(missing ∘ sin) === missing`.

Reply:
My takeaway from https://github.com/JuliaLang/julia/issues/54881 is ot not implement special logic for missing in composed function equality.

Reply:
I think it's possible to support `missing` without any special `missing`-specific logic.